### PR TITLE
Use dd instead of tee in Unix-like operating systems

### PR DESCRIPTION
**Pull request by zeertzjq**
Benefits of using `dd` instead of `tee`:
- Memory usage can be possibly reduced as `dd` can read from the temporary file directly instead of reading from stdin.
- If the entered password is wrong, the file content will not be treated as passwords by `sudo`, so only 1 failed attempt will be counted. (Probably related to #34)
- If `sudo` uses something other than password authentication (like `fprintd` fingerprint authentication), the entered password will not end up at the top of the written file.

### CODE REVIEW

**Comment by lambdalisue** [comment]
Thanks a lot! That's very interesting. I need to try and investigate so please wait for this weekend. Please mention me if you feel I might forget this.

File: autoload/suda.vim [hunk 1]
```diff
@@ -1,3 +1,4 @@
+" vim: et sw=2
```
Comment (lambdalisue) [inline autoload/suda.vim]: **must**

Remove this line

File: autoload/suda.vim [hunk 2]
```diff
@@ -89,19 +90,24 @@ function! suda#write(expr, ...) abort range
...
       " Using a full path for tee command to avoid this problem.
-      let tee_cmd = exepath(tee_cmd)
+      let tee_cmd = exepath('tee')
+      let result = suda#system(
+            \ printf('%s %s', shellescape(tee_cmd), shellescape(path)),
+            \ join(readfile(tempfile, 'b'), "\n")
+            \)
+    else
+      let result = suda#system(printf(
+            \ 'dd if=%s of=%s bs=1048576 status=none',
```
Comment (lambdalisue) [inline autoload/suda.vim]: *should*

What is `bs=1048576`? How did you get the number? Please comment the reason or whatever if that number has some meaning.

Comment (lambdalisue) [inline autoload/suda.vim]: question

I'm sorry but I don't know what `status=none` is. Could you explain the behavior of that?

**Review by lambdalisue — CHANGES_REQUESTED** [review]
I haven't try it but did code review. Please fix **must** and *should*.

File: autoload/suda.vim (hunk 2 above)
Comment (zeertzjq) [reply autoload/suda.vim]: `bs=1048576` is equivalent to `bs=1M` for GNU dd or `bs=1m` for BSD dd. Both `bs=1M` and `bs=1m` are non-POSIX, so I used the number of bytes in 1MiB instead. (See [POSIX man pages](https://www.unix.com/man-page/posix/1posix/dd/))

Comment (zeertzjq) [reply autoload/suda.vim]: `status=none` prevents dd from writing to stderr if no errors occur

File: autoload/suda.vim (hunk 1 above)
Comment (zeertzjq) [reply autoload/suda.vim]: OK

File: autoload/suda.vim (hunk 2 above)
Comment (zeertzjq) [reply autoload/suda.vim]: Oh, it seems that `status=none` is non-POSIX, so I will remove it.

File: autoload/suda.vim [hunk 3]
```diff
@@ -89,19 +89,26 @@ function! suda#write(expr, ...) abort range
...
       " Using a full path for tee command to avoid this problem.
-      let tee_cmd = exepath(tee_cmd)
+      let tee_cmd = exepath('tee')
+      let result = suda#system(
+            \ printf('%s %s', shellescape(tee_cmd), shellescape(path)),
+            \ join(readfile(tempfile, 'b'), "\n")
+            \)
+    else
+      " `bs=1048576` is equivalent to `bs=1M` for GNU dd or `bs=1m` for BSD dd
+      " Both `bs=1M` and `bs=1m` are non-POSIX
```
Comment (lambdalisue) [inline autoload/suda.vim:105]: Nice comment :tada:

**Review by lambdalisue — APPROVED** [review]
Thanks for the fix. It seems the code is OK now so please wait until I test this behavior on this weekend 👍

**Comment by lambdalisue** [comment]
Before
CleanShot 2021-02-20 at 13 34 18

After
CleanShot 2021-02-20 at 13 35 15

Perfect 🎉

**Pull request merged**